crates/saya/core/src/lib.rs

```diff
@@ -81,6 +84,16 @@ impl Saya {
         let poll_interval_secs = 1;
         let mut block = self.config.start_block;
 
+        let (mut previous_block, genesis_state_hash) = if block == 0 {
+            let block = self.provider.fetch_block(0).await?;
+            let genesis_state_hash = block.header.header.state_root;
+            (block, genesis_state_hash)
```

Comment:
```suggestion
            // Genesis block is not proved. We advance to block 1
            // being the next to be processed.
            block += 1;
            (block, genesis_state_hash)
```
If we're starting from genesis, then the first block to expect to be proven is block 1, do you agree?

Reply:
I think this is right, updated